### PR TITLE
ci: improve docker build times using ecr cache

### PR DESCRIPTION
**Pull request by minhlai**
# SC-916

## Proposed changes

- login to ECR and pass CMS and Portal Client repository IDs to docker-compose.yml
- docker-compose.yml will use the cache when available, otherwise will build without it

---

## Setup

You can test that the docker-compose.yml file still works locally. I recommend cloning the [e2e repo](https://github.com/USSF-ORBIT/ussf-portal) and running `yarn services:up` from inside the /e2e folder. This will test the Dockerfile changes with `docker compose`.

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
- [ ] Checked that the E2E test build is not failing

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged

### CODE REVIEW

**Comment by minhlai** [comment]
The ` @github-actions Run E2E tests when changed / run-e2e-tests / run-e2e-tests (pull_request) ` failed to pass because the AWS creds that are present in the Github Actions Secrets of Portal Client repo are not present in this repo. Though they are available as organization secrets. @esacteksab this behavior makes sense to me. The way I architected the reusable workflow call, this workflow will inherit the secrets of the calling repo.

**Review by esacteksab — APPROVED** [review]

**Comment by gidjin** [comment]
> The `@github-actions Run E2E tests when changed / run-e2e-tests / run-e2e-tests (pull_request)` failed to pass because the AWS creds that are present in the Github Actions Secrets of Portal Client repo are not present in this repo. Though they are available as organization secrets. @esacteksab this behavior makes sense to me. The way I architected the reusable workflow call, this workflow will inherit the secrets of the calling repo.

Question: Does this mean that the job will not run in this repo?

File: .github/workflows/run-e2e-tests.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ on:
 jobs:
   run-e2e-tests:
     uses: ./.github/workflows/reusable-run-e2e-tests.yml
+    secrets: inherit
```
Comment (gidjin) [inline .github/workflows/run-e2e-tests.yml:18]: Question: Do need to do this in the cms workflow as well?

Comment (minhlai) [reply .github/workflows/run-e2e-tests.yml:18]: Yes! I totally forgot about the tests being called from the CMS repo as well.

**Comment by minhlai** [comment]
> Question: Does this mean that the job will not run in this repo?

I believe they will. I think I was mistaken about why the Org Secrets were not accessible in this workflow. I now believe they were not accessible because I did not declare that the reusable workflow should inherit the secrets of the caller job. I couldn't find any explicit answer but I think reusable workflows cannot themselves access Github Secrets unless the secrets are inherited or added as inputs. The CMS repo only has Org Secrets and the AWS jobs there work fine.

**Pull request merged**